geomstats/geometry/discrete_curves.py

```diff
@@ -440,3 +440,155 @@ def dist(self, point_a, point_b):
+
+    Parameters
+    ----------
+    ambient_manifold : Manifold
+        Manifold in which curves take values.
+
+    References
+    ----------
+    .. [Sea2011] A. Srivastava, E. Klassen, S. H. Joshi and I. H. Jermyn,
+    "Shape Analysis of Elastic Curves in Euclidean Spaces,"
+    in IEEE Transactions on Pattern Analysis and Machine Intelligence,
+    vol. 33, no. 7, pp. 1415-1428, July 2011.
+    """
+
+    def __init__(self, ambient_manifold):
+        super(ClosedSRVMetric, self).__init__(ambient_manifold)
+
+    def project_srv(self, srv, atol=gs.atol):
+        """Project a point in the srv space into the space of closed curves srv.
+
+        The algorithm is from the paper cited above and modifies the srv
+        iteratively so that G(srv) = (0, ..., 0) with the paper's notation.
+
+        Parameters
+        ----------
+        srv : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        atol : float
+            Tolerance of the projection algorithm.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        proj : array-like, shape=[..., n_sampling_points, ambient_dim]
+        """
+        is_euclidean = isinstance(self.ambient_metric, EuclideanMetric)
+        is_planar = is_euclidean and self.ambient_metric.dim == 2
+
+        if not is_planar:
+            raise AssertionError('The projection is only implemented '
+                                 'for discrete curves embedded in a'
+                                 '2D Euclidean space.')
+
+        n_sampling_points = srv.shape[-2]
+        dim = self.ambient_metric.dim
+        srv_metric = self.l2_metric(n_sampling_points)
+        inner_prod = self.ambient_metric.inner_product
+
+        def g_criterion(srv, srv_norms):
+            return gs.sum(srv * srv_norms[:, None], axis=0)
+
+        initial_norm = srv_metric.norm(srv)
+        proj = srv
+        proj_norms = self.ambient_metric.norm(proj)
+        criteria = atol + 1
+        residual = g_criterion(proj, proj_norms)
+
+        while criteria >= atol:
+            g_jacobian = gs.zeros((dim, dim))
+            for i in range(dim):
+                for j in range(dim):
+                    g_jacobian[i, j] = 3 * inner_prod(proj[:, i], proj[:, j])
+            proj_squared_norm = srv_metric.squared_norm(proj)
+            g_jacobian += proj_squared_norm * gs.array([[1, 0], [0, 1]])
+            beta = gs.linalg.inv(g_jacobian) @ residual
+
+            e_1, e_2 = gs.array([1, 0]), gs.array([0, 1])
+            grad_1 = proj_norms[:, None] * e_1
+            grad_1 = grad_1 + (proj[:, 0] / proj_norms)[:, None] * proj
+            grad_2 = proj_norms[:, None] * e_2
+            grad_2 = grad_2 + (proj[:, 1] / proj_norms)[:, None] * proj
+
+            b_1 = grad_1 / srv_metric.norm(grad_1)
+            b_2 = grad_2 - srv_metric.inner_product(grad_2, b_1) * grad_2
+            b_2 = b_2 / srv_metric.norm(b_2)
+            b = gs.array([b_1, b_2])
```

Comment:
As Nina said, we try to avoid single letter names, so here something like ```basis_vector_1```, ```basis_vector_2```, ```basis``` would be more instructive.